services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -3383,37 +3409,106 @@ private OfflinePushStatusInfo getOffLineJobStatus(
         statusDetails.ifPresent(s -> extraDetails.put(region, leaderControllerUrl + " " + s));
       }
     }
-    // Sort the per-datacenter status in this order, and return the first one in the list
-    // Edge case example: if one cluster is stuck in NOT_CREATED, then
-    // as another cluster goes from PROGRESS to COMPLETED
-    // the aggregate status will go from PROGRESS back down to NOT_CREATED.
-    statuses.sort(Comparator.comparingInt(VeniceHelixAdmin.STATUS_PRIORITIES::indexOf));
-    if (statuses.size() > 0) {
-      currentReturnStatus = statuses.get(0);
+
+    /**
+     * Based on the global information, start determining the final status
+     */
+    ExecutionStatus currentReturnStatus = ExecutionStatus.NEW;
+    StringBuilder currentReturnStatusDetails = new StringBuilder();
+    List<ExecutionStatus> sortedStatuses = new ArrayList<>();
+
+    if (StringUtils.isEmpty(targetedRegions)) {
+      // Sort the per-datacenter status in this order, and return the first one in the list
+      // Edge case example: if one cluster is stuck in NOT_CREATED, then
+      // as another cluster goes from PROGRESS to COMPLETED
+      // the aggregate status will go from PROGRESS back down to NOT_CREATED.
+      sortedStatuses = statuses.values()
+          .stream()
+          .sorted(Comparator.comparingInt(VeniceHelixAdmin.STATUS_PRIORITIES::indexOf))
+          .collect(Collectors.toList());
+
+      currentReturnStatus = getFinalReturnStatus(sortedStatuses, childRegions, failCount, currentReturnStatusDetails);
+    } else {
+      // we use the same logic as above, but only consider the targeted regions
+      Set<String> targetedRegionsSet = RegionUtils.parseRegionsFilterList(targetedRegions);
+      int targetedRegionFailCount = 0;
+      for (String region: targetedRegionsSet) {
+        ExecutionStatus status = statuses.get(region);
+        /**
+         * We consider failures for the following cases:
+         * 1. the provided region isn't available for the cluster, in other words, invalid targeted region is provided.
+         * 2. the provided region has troubles to query the job status.
+         */
+        if (status != null) {
+          sortedStatuses.add(status);
+          if (status.equals(ExecutionStatus.UNKNOWN)) {
+            targetedRegionFailCount += 1;
+          }
+        } else {
+          targetedRegionFailCount += 1;
+        }
+      }
+      sortedStatuses = sortedStatuses.stream()
+          .sorted(Comparator.comparingInt(VeniceHelixAdmin.STATUS_PRIORITIES::indexOf))
+          .collect(Collectors.toList());
+
+      currentReturnStatusDetails.append("This is targeted region push and following regions ")
+          .append(targetedRegions)
+          .append(" are expected to be COMPLETED. ");
+      currentReturnStatus =
+          getFinalReturnStatus(sortedStatuses, targetedRegionsSet, targetedRegionFailCount, currentReturnStatusDetails);
```

Comment:
I am hoping that If we can move `currentReturnStatusDetails` before @L3472, then the remaining part becomes a common code for L3446-L3451, so that we can move it outside the branch and change would be much simpler.

Reply:
For some weird reasons, github is not showing you the latest changes. I've done it already..